agent/log-drop/log-drop_test.go

```diff
@@ -0,0 +1,35 @@
+package log_drop
+
+import (
+	"context"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func TestNewLogDrop(t *testing.T) {
+	mockLogger := NewMockLogger(t)
+	mockLogger.On("Info", "test Log", []interface{}{"val", 0}).Return()
+	ld := NewLogDrop(context.Background(), "test", 10, mockLogger, func(_ Log) {})
+	require.NotNil(t, ld)
+	ld.Info("test Log", "val", 0)
+	time.Sleep(10 * time.Millisecond)
+	mockLogger.AssertNumberOfCalls(t, "Info", 1)
+}
```

Comment:
I think we can remove the sleep here with some clever mock/channel tricks 🤓

```suggestion
func TestNewLogDrop(t *testing.T) {
	callCh := make(chan struct{})
	mockLogger := NewMockLogger(t)
	mockLogger.On("Info", "test Log", []interface{}{"val", 0}).
		Run(func(mock.Arguments) { close(callCh) }).
		Return()
	
	ld := NewLogDrop(context.Background(), "test", 10, mockLogger, func(_ Log) {})
	require.NotNil(t, ld)
	
	ld.Info("test Log", "val", 0)
	
	select {
	case <-callCh:
	case <-time.After(100 * time.Millisecond):
		t.Fatal("timeout waiting for logger to be called")
	}
}
```